### PR TITLE
Fix output extension if user has no transcoding permission

### LINKED ISSUE #7901: [Issue]: Remux makes ffmpeg fail with "Invalid argument"
### Please describe your bug

1. Have a file with a supported video track, but an unsupported audio track
2. Play it with a client (Android TV in my case)
3. Client fails to play. Logs contain an "invalid argument" error from ffmpeg, seemingly due to missing extension on the output file.

---

Full transcodes play normally, example command line:

```
/run/current-system/sw/bin/ffmpeg -analyzeduration 200M -ss 00:09:53.384 -init_hw_device vaapi=va:,driver=iHD,kernel_driver=i915 -init_hw_device qsv=qs@va -init_hw_device opencl=ocl@va -filter_hw_device qs -hwaccel vaapi -hwaccel_output_format vaapi -autorotate 0 -canvas_size 2592x1080 -i file:"/mnt/media/Movies/John Wick Chapter 3 - Parabellum (2019)/John.Wick.Chapter.3.Parabellum.2019.MULTi.UHD.BluRay.2160p.HDR.Atmos.7.1.HEVC-DDR.mkv" -map_metadata -1 -map_chapters -1 -threads 0 -map 0:0 -map 0:1 -codec:v:0 hevc_qsv -tag:v:0 hvc1 -preset 7 -look_ahead 0 -b:v 20662786 -maxrate 20662786 -bufsize 41325572 -profile:v:0 main -level 50 -g:v:0 72 -keyint_min:v:0 72 -filter_complex "[0:5]scale=flags=fast_bilinear,format=bgra,hwupload=extra_hw_frames=64[sub];[0:0]setparams=color_primaries=bt2020:color_trc=smpte2084:colorspace=bt2020nc,hwmap=derive_device=opencl,tonemap_opencl=format=nv12:p=bt709:t=bt709:m=bt709:tonemap=bt2390:peak=100:desat=0,hwmap=derive_device=qsv:reverse=1:extra_hw_frames=16,format=qsv[main];[main][sub]overlay_qsv=eof_action=endall:shortest=1:repeatlast=0:w=3840:h=1600" -start_at_zero -codec:a:0 ac3 -ac 6 -ab 640000 -copyts -avoid_negative_ts disabled -max_muxing_queue_size 2048 -f hls -max_delay 5000000 -hls_time 3 -hls_segment_type mpegts -start_number 75 -hls_segment_filename "/var/lib/jellyfin/transcodes/97042807d79d30c8fcf6d05c06086be1%d.ts" -hls_playlist_type vod -hls_list_size 0 -y "/var/lib/jellyfin/transcodes/97042807d79d30c8fcf6d05c06086be1.m3u8"
```

### Jellyfin Version

10.8.0

### if other:

_No response_

### Environment

```markdown
- OS: NixOS 22.05
- Virtualization: none
- Clients: Android TV
- Browser: N/A
- FFmpeg Version: 5.0.1-5 (Jellyfin)
- Playback Method: Remux
- Hardware Acceleration: VAAPI (Intel)
- Plugins: none
- Reverse Proxy: none
- Base URL: none
- Networking: host
- Storage: local
```

### Jellyfin logs

```shell
[2022-06-12 19:36:26.138 +00:00] [INF] [23] Jellyfin.Api.Helpers.TranscodingJobHelper: "/run/current-system/sw/bin/ffmpeg" "-analyzeduration 200M -fflags +genpts  -i file:\"/mnt/media/Movies/Joker (2019)/Joker.2019.2160p.UHD.BluRay.x265-AAAUHD.mkv\" -map 0:0 -map 0:1 -map -0:s -codec:v:0 copy -map_metadata -1 -map_chapters -1 -threads 0 -codec:a:0 copy -y \"/var/lib/jellyfin/transcodes/686d1ce8ad5d60896e03abbb3f4d14b1\""
[2022-06-12 19:36:26.235 +00:00] [ERR] [23] Jellyfin.Api.Helpers.TranscodingJobHelper: FFmpeg exited with code 1
[2022-06-12 19:36:26.308 +00:00] [ERR] [23] Jellyfin.Server.Middleware.ExceptionMiddleware: Error processing request. URL "GET" "/videos/2b367999-ace0-2881-a456-1f3bc8d06fe9/stream".
MediaBrowser.Common.FfmpegException: FFmpeg exited with code 1
   at Jellyfin.Api.Helpers.TranscodingJobHelper.StartFfMpeg(StreamState state, String outputPath, String commandLineArguments, HttpRequest request, TranscodingJobType transcodingJobType, CancellationTokenSource cancellationTokenSource, String workingDirectory) in /build/source/Jellyfin.Api/Helpers/TranscodingJobHelper.cs:line 645
   at Jellyfin.Api.Helpers.FileStreamResponseHelpers.GetTranscodedFile(StreamState state, Boolean isHeadRequest, HttpContext httpContext, TranscodingJobHelper transcodingJobHelper, String ffmpegCommandLineArguments, TranscodingJobType transcodingJobType, CancellationTokenSource cancellationTokenSource) in /build/source/Jellyfin.Api/Helpers/FileStreamResponseHelpers.cs:line 103
   at Jellyfin.Api.Controllers.VideosController.GetVideoStream(Guid itemId, String container, Nullable`1 static, String params, String tag, String deviceProfileId, String playSessionId, String segmentContainer, Nullable`1 segmentLength, Nullable`1 minSegments, String mediaSourceId, String deviceId, String audioCodec, Nullable`1 enableAutoStreamCopy, Nullable`1 allowVideoStreamCopy, Nullable`1 allowAudioStreamCopy, Nullable`1 breakOnNonKeyFrames, Nullable`1 audioSampleRate, Nullable`1 maxAudioBitDepth, Nullable`1 audioBitRate, Nullable`1 audioChannels, Nullable`1 maxAudioChannels, String profile, String level, Nullable`1 framerate, Nullable`1 maxFramerate, Nullable`1 copyTimestamps, Nullable`1 startTimeTicks, Nullable`1 width, Nullable`1 height, Nullable`1 maxWidth, Nullable`1 maxHeight, Nullable`1 videoBitRate, Nullable`1 subtitleStreamIndex, Nullable`1 subtitleMethod, Nullable`1 maxRefFrames, Nullable`1 maxVideoBitDepth, Nullable`1 requireAvc, Nullable`1 deInterlace, Nullable`1 requireNonAnamorphic, Nullable`1 transcodingMaxAudioChannels, Nullable`1 cpuCoreLimit, String liveStreamId, Nullable`1 enableMpegtsM2TsMode, String videoCodec, String subtitleCodec, String transcodeReasons, Nullable`1 audioStreamIndex, Nullable`1 videoStreamIndex, Nullable`1 context, Dictionary`2 streamOptions) in /build/source/Jellyfin.Api/Controllers/VideosController.cs:line 503
   at lambda_method1174(Closure , Object )
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Jellyfin.Server.Middleware.ServerStartupMessageMiddleware.Invoke(HttpContext httpContext, IServerApplicationHost serverApplicationHost, ILocalizationManager localizationManager) in /build/source/Jellyfin.Server/Middleware/ServerStartupMessageMiddleware.cs:line 49
   at Jellyfin.Server.Middleware.WebSocketHandlerMiddleware.Invoke(HttpContext httpContext, IWebSocketManager webSocketManager) in /build/source/Jellyfin.Server/Middleware/WebSocketHandlerMiddleware.cs:line 38
   at Jellyfin.Server.Middleware.IpBasedAccessValidationMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager) in /build/source/Jellyfin.Server/Middleware/IpBasedAccessValidationMiddleware.cs:line 48
   at Jellyfin.Server.Middleware.LanFilteringMiddleware.Invoke(HttpContext httpContext, INetworkManager networkManager, IServerConfigurationManager serverConfigurationManager) in /build/source/Jellyfin.Server/Middleware/LanFilteringMiddleware.cs:line 43
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.QueryStringDecodingMiddleware.Invoke(HttpContext httpContext) in /build/source/Jellyfin.Server/Middleware/QueryStringDecodingMiddleware.cs:line 37
   at Swashbuckle.AspNetCore.ReDoc.ReDocMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jellyfin.Server.Middleware.RobotsRedirectionMiddleware.Invoke(HttpContext httpContext) in /build/source/Jellyfin.Server/Middleware/RobotsRedirectionMiddleware.cs:line 45
   at Jellyfin.Server.Middleware.LegacyEmbyRouteRewriteMiddleware.Invoke(HttpContext httpContext) in /build/source/Jellyfin.Server/Middleware/LegacyEmbyRouteRewriteMiddleware.cs:line 52
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)
   at Jellyfin.Server.Middleware.ResponseTimeMiddleware.Invoke(HttpContext context) in /build/source/Jellyfin.Server/Middleware/ResponseTimeMiddleware.cs:line 64
   at Jellyfin.Server.Middleware.ExceptionMiddleware.Invoke(HttpContext context) in /build/source/Jellyfin.Server/Middleware/ExceptionMiddleware.cs:line 101
[2022-06-12 19:36:27.225 +00:00] [INF] [36] Jellyfin.Api.Helpers.TranscodingJobHelper: Deleting partial stream file(s) "/var/lib/jellyfin/transcodes/686d1ce8ad5d60896e03abbb3f4d14b1"
[2022-06-12 19:36:27.225 +00:00] [INF] [36] Jellyfin.Api.Helpers.TranscodingJobHelper: Deleting partial stream file(s) "/var/lib/jellyfin/transcodes/686d1ce8ad5d60896e03abbb3f4d14b1"
[2022-06-12 19:36:27.225 +00:00] [INF] [36] Jellyfin.Api.Helpers.TranscodingJobHelper: Deleting partial stream file(s) "/var/lib/jellyfin/transcodes/686d1ce8ad5d60896e03abbb3f4d14b1"
[2022-06-12 19:36:27.225 +00:00] [INF] [36] Jellyfin.Api.Helpers.TranscodingJobHelper: Deleting partial stream file(s) "/var/lib/jellyfin/transcodes/686d1ce8ad5d60896e03abbb3f4d14b1"
[2022-06-12 19:36:28.728 +00:00] [INF] [35] Emby.Server.Implementations.Session.SessionManager: Playback stopped reported by app "Android TV" "0.13.6" playing "Joker". Stopped at "0" ms
[2022-06-12 19:36:28.742 +00:00] [WRN] [35] Jellyfin.Server.Middleware.ResponseTimeMiddleware: Slow HTTP Response from "http://calor:8096/Sessions/Playing/Stopped" to "192.168.1.173" in 0:00:01.5210209 with Status Code 204
```

### FFmpeg logs

```shell
/videos/2b367999-ace0-2881-a456-1f3bc8d06fe9/stream

{"Protocol":0,"Id":"2b367999ace02881a4561f3bc8d06fe9","Path":"/mnt/media/Movies/Joker (2019)/Joker.2019.2160p.UHD.BluRay.x265-AAAUHD.mkv","EncoderPath":null,"EncoderProtocol":null,"Type":0,"Container":"mkv,webm","Size":17411473408,"Name":"Joker.2019.2160p.UHD.BluRay.x265-AAAUHD","IsRemote":false,"ETag":"bccbfac7ed9682a5fa2fc7d490f4ca5c","RunTimeTicks":73091203072,"ReadAtNativeFramerate":false,"IgnoreDts":false,"IgnoreIndex":false,"GenPtsInput":false,"SupportsTranscoding":true,"SupportsDirectStream":true,"SupportsDirectPlay":true,"IsInfiniteStream":false,"RequiresOpening":false,"OpenToken":null,"RequiresClosing":false,"LiveStreamId":null,"BufferMs":null,"RequiresLooping":false,"SupportsProbing":true,"VideoType":0,"IsoType":null,"Video3DFormat":null,"MediaStreams":[{"Codec":"hevc","CodecTag":null,"Language":null,"ColorRange":null,"ColorSpace":"bt2020nc","ColorTransfer":"smpte2084","ColorPrimaries":"bt2020","Comment":null,"TimeBase":"1/1000","CodecTimeBase":null,"Title":"Joker.2019.2160p.UHD.BluRay.x265-AAAUHD","VideoRange":"HDR","LocalizedUndefined":null,"LocalizedDefault":null,"LocalizedForced":null,"LocalizedExternal":null,"DisplayTitle":"Joker.2019.2160p.UHD.BluRay.x265-AAAUHD - 4K - HEVC - HDR","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":null,"BitRate":19057257,"BitDepth":null,"RefFrames":1,"PacketLength":null,"Channels":null,"SampleRate":null,"IsDefault":true,"IsForced":false,"Height":2080,"Width":3840,"AverageFrameRate":23.976025,"RealFrameRate":23.976025,"Profile":"Main 10","Type":1,"AspectRatio":"1.85:1","Index":0,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":"yuv420p10le","Level":153,"IsAnamorphic":null},{"Codec":"truehd","CodecTag":null,"Language":"eng","ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":null,"TimeBase":"1/1000","CodecTimeBase":null,"Title":"English Dolby TrueHD with Dolby Atmos (TrueHD)","VideoRange":null,"LocalizedUndefined":null,"LocalizedDefault":null,"LocalizedForced":null,"LocalizedExternal":null,"DisplayTitle":"English Dolby TrueHD with Dolby Atmos (TrueHD) - 7.1 - Default","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":"7.1","BitRate":3819392,"BitDepth":24,"RefFrames":null,"PacketLength":null,"Channels":8,"SampleRate":48000,"IsDefault":true,"IsForced":false,"Height":null,"Width":null,"AverageFrameRate":null,"RealFrameRate":null,"Profile":null,"Type":0,"AspectRatio":null,"Index":1,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":null,"Level":0,"IsAnamorphic":null},{"Codec":"ac3","CodecTag":null,"Language":"eng","ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":null,"TimeBase":"1/1000","CodecTimeBase":null,"Title":"English Dolby Digital (AC3)","VideoRange":null,"LocalizedUndefined":null,"LocalizedDefault":null,"LocalizedForced":null,"LocalizedExternal":null,"DisplayTitle":"English Dolby Digital (AC3) - 5.1","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":"5.1","BitRate":640000,"BitDepth":null,"RefFrames":null,"PacketLength":null,"Channels":6,"SampleRate":48000,"IsDefault":false,"IsForced":false,"Height":null,"Width":null,"AverageFrameRate":null,"RealFrameRate":null,"Profile":null,"Type":0,"AspectRatio":null,"Index":2,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":null,"Level":0,"IsAnamorphic":null},{"Codec":"subrip","CodecTag":null,"Language":"eng","ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":null,"TimeBase":"1/1000","CodecTimeBase":null,"Title":"English SDH","VideoRange":null,"LocalizedUndefined":"Undefined","LocalizedDefault":"Default","LocalizedForced":"Forced","LocalizedExternal":"External","DisplayTitle":"English SDH - SUBRIP","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":null,"BitRate":null,"BitDepth":null,"RefFrames":null,"PacketLength":null,"Channels":null,"SampleRate":null,"IsDefault":false,"IsForced":false,"Height":null,"Width":null,"AverageFrameRate":null,"RealFrameRate":null,"Profile":null,"Type":2,"AspectRatio":null,"Index":3,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":true,"SupportsExternalStream":true,"Path":null,"PixelFormat":null,"Level":0,"IsAnamorphic":null},{"Codec":"PGSSUB","CodecTag":null,"Language":"eng","ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":null,"TimeBase":"1/1000","CodecTimeBase":null,"Title":"English PGS","VideoRange":null,"LocalizedUndefined":"Undefined","LocalizedDefault":"Default","LocalizedForced":"Forced","LocalizedExternal":"External","DisplayTitle":"English PGS - PGSSUB","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":null,"BitRate":null,"BitDepth":null,"RefFrames":null,"PacketLength":null,"Channels":null,"SampleRate":null,"IsDefault":false,"IsForced":false,"Height":null,"Width":null,"AverageFrameRate":null,"RealFrameRate":null,"Profile":null,"Type":2,"AspectRatio":null,"Index":4,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":null,"Level":0,"IsAnamorphic":null},{"Codec":"srt","CodecTag":null,"Language":"lit","ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"Comment":null,"TimeBase":null,"CodecTimeBase":null,"Title":null,"VideoRange":null,"LocalizedUndefined":"Undefined","LocalizedDefault":"Default","LocalizedForced":"Forced","LocalizedExternal":"External","DisplayTitle":"Lithuanian - SRT - External","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":null,"BitRate":null,"BitDepth":null,"RefFrames":null,"PacketLength":null,"Channels":null,"SampleRate":null,"IsDefault":false,"IsForced":false,"Height":null,"Width":null,"AverageFrameRate":null,"RealFrameRate":null,"Profile":null,"Type":2,"AspectRatio":null,"Index":5,"Score":null,"IsExternal":true,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":true,"SupportsExternalStream":true,"Path":"/mnt/media/Movies/Joker (2019)/Joker.2019.2160p.UHD.BluRay.x265-AAAUHD.lt.srt","PixelFormat":null,"Level":null,"IsAnamorphic":null}],"MediaAttachments":[],"Formats":[],"Bitrate":23516649,"Timestamp":null,"RequiredHttpHeaders":{},"TranscodingUrl":null,"TranscodingSubProtocol":null,"TranscodingContainer":null,"AnalyzeDurationMs":null,"DefaultAudioStreamIndex":null,"DefaultSubtitleStreamIndex":null}

/run/current-system/sw/bin/ffmpeg -analyzeduration 200M -fflags +genpts  -i file:"/mnt/media/Movies/Joker (2019)/Joker.2019.2160p.UHD.BluRay.x265-AAAUHD.mkv" -map 0:0 -map 0:1 -map -0:s -codec:v:0 copy -map_metadata -1 -map_chapters -1 -threads 0 -codec:a:0 copy -y "/var/lib/jellyfin/transcodes/686d1ce8ad5d60896e03abbb3f4d14b1"


ffmpeg version 5.0.1 Copyright (c) 2000-2022 the FFmpeg developers
  built with gcc 11.3.0 (GCC)
  configuration: --disable-static --prefix=/nix/store/4cspn4sjk8sqidjmq9csbwincfsnq2q4-jellyfin-ffmpeg-5.0.1-5 --target_os=linux --arch=x86_64 --enable-gpl --enable-version3 --disable-nonfree --enable-shared --enable-pic --disable-small --enable-runtime-cpudetect --disable-lto --enable-gray --enable-swscale-alpha --enable-hardcoded-tables --enable-safe-bitstream-reader --enable-pthreads --disable-w32threads --disable-os2threads --enable-network --enable-pixelutils --enable-ffmpeg --enable-ffplay --enable-ffprobe --enable-avcodec --enable-avdevice --enable-avfilter --enable-avformat --enable-avutil --enable-postproc --enable-swresample --enable-swscale --enable-doc --disable-htmlpages --enable-manpages --disable-podpages --disable-txtpages --enable-bzlib --enable-libcelt --enable-libdav1d --disable-libfdk-aac --disable-libflite --enable-fontconfig --enable-libfreetype --enable-frei0r --enable-libfribidi --enable-libgme --enable-gnutls --enable-libgsm --enable-ladspa --enable-libmp3lame --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libdc1394 --enable-libdrm --enable-iconv --enable-libjack --enable-libmfx --enable-libmodplug --enable-libmysofa --enable-libopus --enable-librsvg --enable-libsrt --enable-libssh --enable-libtheora --enable-libv4l2 --enable-vaapi --enable-vdpau --enable-libvorbis --enable-libvmaf --enable-libvpx --enable-libwebp --enable-xlib --enable-libxcb --enable-libxcb-shm --enable-libxcb-xfixes --enable-libxcb-shape --enable-libxml2 --enable-lzma --enable-nvenc --enable-openal --enable-opencl --enable-libopencore-amrnb --disable-opengl --enable-libopenh264 --enable-libopenjpeg --disable-openssl --enable-libpulse --disable-librav1e --enable-libsvtav1 --disable-librtmp --enable-sdl2 --enable-libsoxr --enable-libspeex --enable-libvidstab --enable-libvo-amrwbenc --enable-libx264 --enable-libx265 --enable-libxavs --enable-libxvid --enable-libzmq --enable-libzimg --enable-zlib --enable-vulkan --enable-libglslang --enable-libsmbclient --disable-debug --enable-optimizations --disable-extra-warnings --disable-stripping
  libavutil      57. 17.100 / 57. 17.100
  libavcodec     59. 18.100 / 59. 18.100
  libavformat    59. 16.100 / 59. 16.100
  libavdevice    59.  4.100 / 59.  4.100
  libavfilter     8. 24.100 /  8. 24.100
  libswscale      6.  4.100 /  6.  4.100
  libswresample   4.  3.100 /  4.  3.100
  libpostproc    56.  3.100 / 56.  3.100
Input #0, matroska,webm, from 'file:/mnt/media/Movies/Joker (2019)/Joker.2019.2160p.UHD.BluRay.x265-AAAUHD.mkv':
  Metadata:
    encoder         : libebml v1.3.9 + libmatroska v1.5.2
    creation_time   : 2019-12-20T00:41:28.000000Z
  Duration: 02:01:49.12, start: 0.000000, bitrate: 19057 kb/s
  Chapters:
    Chapter #0:0: start 0.000000, end 628.795000
      Metadata:
        title           : Chapter 01
    Chapter #0:1: start 628.795000, end 1261.302000
      Metadata:
        title           : Chapter 02
    Chapter #0:2: start 1261.302000, end 1724.973000
      Metadata:
        title           : Chapter 03
    Chapter #0:3: start 1724.973000, end 2294.501000
      Metadata:
        title           : Chapter 04
    Chapter #0:4: start 2294.501000, end 2781.570000
      Metadata:
        title           : Chapter 05
    Chapter #0:5: start 2781.570000, end 3379.293000
      Metadata:
        title           : Chapter 06
    Chapter #0:6: start 3379.293000, end 3881.169000
      Metadata:
        title           : Chapter 07
    Chapter #0:7: start 3881.169000, end 4541.746000
      Metadata:
        title           : Chapter 08
    Chapter #0:8: start 4541.746000, end 5176.046000
      Metadata:
        title           : Chapter 09
    Chapter #0:9: start 5176.046000, end 5822.608000
      Metadata:
        title           : Chapter 10
    Chapter #0:10: start 5822.608000, end 6350.386000
      Metadata:
        title           : Chapter 11
    Chapter #0:11: start 6350.386000, end 6944.646000
      Metadata:
        title           : Chapter 12
    Chapter #0:12: start 6944.646000, end 7309.120000
      Metadata:
        title           : Chapter 13
  Stream #0:0: Video: hevc (Main 10), yuv420p10le(tv, bt2020nc/bt2020/smpte2084), 3840x2080 [SAR 1:1 DAR 24:13], 23.98 fps, 23.98 tbr, 1k tbn (default)
    Metadata:
      title           : Joker.2019.2160p.UHD.BluRay.x265-AAAUHD
      BPS-eng         : 14495468
      DURATION-eng    : 02:01:49.094000000
      NUMBER_OF_FRAMES-eng: 175243
      NUMBER_OF_BYTES-eng: 13243592829
      _STATISTICS_WRITING_APP-eng: mkvmerge v40.0.0 ('Old Town Road + Pony') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2019-12-20 00:41:28
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
  Stream #0:1(eng): Audio: truehd, 48000 Hz, 7.1, s32 (24 bit) (default)
    Metadata:
      title           : English Dolby TrueHD with Dolby Atmos (TrueHD)
      BPS-eng         : 3819392
      DURATION-eng    : 02:01:49.094000000
      NUMBER_OF_FRAMES-eng: 8770913
      NUMBER_OF_BYTES-eng: 3489537684
      _STATISTICS_WRITING_APP-eng: mkvmerge v40.0.0 ('Old Town Road + Pony') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2019-12-20 00:41:28
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
  Stream #0:2(eng): Audio: ac3, 48000 Hz, 5.1(side), fltp, 640 kb/s
    Metadata:
      title           : English Dolby Digital (AC3)
      BPS-eng         : 640000
      DURATION-eng    : 02:01:49.120000000
      NUMBER_OF_FRAMES-eng: 228410
      NUMBER_OF_BYTES-eng: 584729600
      _STATISTICS_WRITING_APP-eng: mkvmerge v40.0.0 ('Old Town Road + Pony') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2019-12-20 00:41:28
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
  Stream #0:3(eng): Subtitle: subrip
    Metadata:
      title           : English SDH
      BPS-eng         : 55
      DURATION-eng    : 01:58:14.086000000
      NUMBER_OF_FRAMES-eng: 1735
      NUMBER_OF_BYTES-eng: 49022
      _STATISTICS_WRITING_APP-eng: mkvmerge v40.0.0 ('Old Town Road + Pony') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2019-12-20 00:41:28
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
  Stream #0:4(eng): Subtitle: hdmv_pgs_subtitle, 1920x1080
    Metadata:
      title           : English PGS
      BPS-eng         : 33769
      DURATION-eng    : 01:58:14.087000000
      NUMBER_OF_FRAMES-eng: 3480
      NUMBER_OF_BYTES-eng: 29945618
      _STATISTICS_WRITING_APP-eng: mkvmerge v40.0.0 ('Old Town Road + Pony') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2019-12-20 00:41:28
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
[NULL @ 0x110df00] Unable to find a suitable output format for '/var/lib/jellyfin/transcodes/686d1ce8ad5d60896e03abbb3f4d14b1'
/var/lib/jellyfin/transcodes/686d1ce8ad5d60896e03abbb3f4d14b1: Invalid argument
```

### Please attach any browser or client logs here

_No response_

### Please attach any screenshots here

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by nyanmisaka**
**Changes**
- Fix output extension if user has no transcoding permission

**Issues**
Fixes #7901

### CODE REVIEW

**Review by crobibero — APPROVED** [review]

File: Jellyfin.Api/Helpers/StreamingHelpers.cs [hunk 1]
```diff
@@ -474,6 +475,17 @@ private static void AddTimeSeekResponseHeaders(StreamState state, IHeaderDiction
                 }
             }
 
+            // Fallback to the container of mediaSource
+            if (mediaSource != null && !string.IsNullOrEmpty(mediaSource.Container))
```
Comment (cvium) [inline Jellyfin.Api/Helpers/StreamingHelpers.cs:479]:
```suggestion
            if (!string.IsNullOrEmpty(mediaSource?.Container))
```

File: Jellyfin.Api/Helpers/StreamingHelpers.cs [hunk 2]
```diff
@@ -474,6 +475,17 @@ private static void AddTimeSeekResponseHeaders(StreamState state, IHeaderDiction
...
+            {
+                var containers = mediaSource.Container.Split(',', StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
```
Comment (cvium) [inline Jellyfin.Api/Helpers/StreamingHelpers.cs:481]: Why split on `','`?

Comment (nyanmisaka) [reply Jellyfin.Api/Helpers/StreamingHelpers.cs:481]: Yes. `mkv, webm`

File: Jellyfin.Api/Helpers/StreamingHelpers.cs [hunk 3]
```diff
@@ -474,6 +475,17 @@ private static void AddTimeSeekResponseHeaders(StreamState state, IHeaderDiction
...
 
+            // Fallback to the container of mediaSource
+            if (!string.IsNullOrEmpty(mediaSource?.Container))
+            {
+                var containers = mediaSource.Container.Split(',', StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+
+                if (containers.Length > 0)
+                {
+                    return '.' + containers[0];
+                }
```
Comment (cvium) [inline Jellyfin.Api/Helpers/StreamingHelpers.cs:486]: I think this is identical
```suggestion
                var idx = mediaSource.Container.IndexOf(',');
                return '.' + (idx == -1 ? mediaSource.Container : mediaSource.Container[..idx]);
```

Comment (nyanmisaka) [reply Jellyfin.Api/Helpers/StreamingHelpers.cs]: It works.

File: Jellyfin.Api/Helpers/StreamingHelpers.cs [hunk 4]
```diff
@@ -474,6 +475,13 @@ private static void AddTimeSeekResponseHeaders(StreamState state, IHeaderDiction
                 }
             }
 
+            // Fallback to the container of mediaSource
+            if (!string.IsNullOrEmpty(mediaSource?.Container))
+            {
+                var idx = mediaSource.Container.IndexOf(',', StringComparison.OrdinalIgnoreCase);
+                return '.' + (idx == -1 ? mediaSource.Container : mediaSource.Container[..idx]);
```
Comment (cvium) [inline Jellyfin.Api/Helpers/StreamingHelpers.cs:482]: Do you think Trim is needed?

**Review by cvium — APPROVED** [review]

File: Jellyfin.Api/Helpers/StreamingHelpers.cs (hunk 4 above)
Comment (nyanmisaka) [reply Jellyfin.Api/Helpers/StreamingHelpers.cs]: I added it just in case.

**Pull request merged**